### PR TITLE
Storage Index

### PR DESCRIPTION
**Pull request by IljaN**

### CODE REVIEW

File: accounts/pkg/service/v0/service.go [hunk 1]
```diff
@@ -72,6 +66,41 @@ func New(opts ...Option) (s *Service, err error) {
...
+	// Accounts
+
+	if err := idx.AddIndex(&proto.Account{}, "DisplayName", "Id", "accounts", "unique"); err != nil {
+		return nil, err
+	}
+	if err := idx.AddIndex(&proto.Account{}, "Mail", "Id", "accounts", "unique"); err != nil {
+		return nil, err
+	}
+
+	if err := idx.AddIndex(&proto.Account{}, "OnPremisesSamAccountName", "Id", "accounts", "non_unique"); err != nil {
```
Comment (kulmann) [inline accounts/pkg/service/v0/service.go:82]: This needs to be a `unique` index.

File: accounts/pkg/service/v0/service.go [hunk 2]
```diff
@@ -72,6 +66,41 @@ func New(opts ...Option) (s *Service, err error) {
 	return
 }
 
+func (s Service) buildIndex() (*indexer.Indexer, error) {
+	s.Config.Repo.CS3.JWTSecret = "Pive-Fumkiu4"
+	idx := indexer.CreateIndexer(s.Config)
+
+	// Accounts
+
+	if err := idx.AddIndex(&proto.Account{}, "DisplayName", "Id", "accounts", "unique"); err != nil {
```
Comment (kulmann) [inline accounts/pkg/service/v0/service.go:75]: This needs to be a `non_unique` index.

File: accounts/pkg/service/v0/service.go (hunk 1 above)
Comment (IljaN) [reply accounts/pkg/service/v0/service.go:82]: :+1:

File: accounts/pkg/service/v0/service.go (hunk 2 above)
Comment (IljaN) [reply accounts/pkg/service/v0/service.go:75]: :+1:

**Review by kulmann — CHANGES_REQUESTED** [review]
So cool 😍 Some comments for tiny improvements. 😃

**Comment by IljaN** [comment]
Merged via https://github.com/owncloud/ocis/pull/709

**Pull request closed without merging**